app/routes.py

```diff
@@ -1 +1,311 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, jsonify, request, abort, make_response
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+import requests
+
+# BLUEPRINTS #
+task_bp = Blueprint("tasks", __name__, url_prefix = "/tasks")
+goal_bp = Blueprint("goals", __name__, url_prefix = "/goals")
+
+# HELPER FUNCTIONS #
+def validate_model(cls, model_id):
+    try: 
+        model_id = int(model_id)
+    except ValueError:
+        abort(make_response({"message": f"{model_id} is not a valid type. A {(type(model_id))} data type was provided. Must be a valid integer data type."},400))
+    
+    model = cls.query.get(model_id)
+    if not model:
+        abort(make_response({"message" : f"{cls.__name__} {model_id} does not exist"},404))
+    
+    return model
+
+def slack_bot_notification(task):
+# SLACK IMPLEMENTATION USING HTTP REQUESTS
+# https://stackoverflow.com/questions/63899742/how-to-use-mock-in-request-post-to-an-external-api
+    import os
+
+    api_url = "https://slack.com/api/chat.postMessage"
+    slack_token = os.environ.get("SLACK_WEB_API_KEY")
+    headers = {"Authorization": slack_token}
+    body = {
+        "channel": "task-notifications",
+        "text": f"Someone just completed the task {task.title}" 
+        }
+    response = requests.post(api_url, headers=headers, params=body)
+
+
+# POST METHOD - create task
+@task_bp.route("", methods = ["POST"])
+def create_task():
+    request_body = request.get_json()
+    try:
+        new_task = Task.from_dict(request_body)
+    except:
+        return jsonify({"details": "Invalid data"}),400
+    
+    db.session.add(new_task)
```

Comment:
Great job using `db.session.<method-name>`. Session gives us access to the follow:

- `db` is our app’s instance of SQLAlchemy.
- `session` represents our active database connection.
- By referencing `db.session` we can use SQLAlchemy’s methods to perform tasks like:
    - committing a change to a model
    - storing a new record of a model
    - deleting a record.
- By referencing `db.session.add()` you are able to use the SQLAlchemy method to store a new record of the task model